pkg/models/application.go

```diff
@@ -0,0 +1,43 @@
+package models
+
+import (
+	"strings"
+
+	"github.com/weaveworks/weave-gitops/pkg/gitproviders"
+)
+
+type AutomationType string
+
+type ConfigType string
+
+const (
+	AutomationTypeHelm      AutomationType = "helm"
+	AutomationTypeKustomize AutomationType = "kustomize"
+
+	ConfigTypeUserRepo ConfigType = ""
+	ConfigTypeNone     ConfigType = "NONE"
+)
+
+type SourceType string
+
+const (
+	SourceTypeGit  SourceType = "git"
+	SourceTypeHelm SourceType = "helm"
+)
+
+type Application struct {
+	Name                string
+	Namespace           string
+	SourceURL           gitproviders.RepoURL
```

Comment:
This won't work in all cases because we have helm repository sources which have no git provider info. The RepoURL creation will fail.

Reply:
Good catch. From standup, my suggestion would be a new struct field on `models.Application` called `HelmChartURL` instead of trying to fit both `Git` and `Helm` URLs on the same field.

Reply:
Added a HelmSourceURL and changed SourceURL to GitSourceURL.